core/src/Call.cpp

```diff
@@ -55,20 +66,33 @@ bool Call::operator()(unsigned int func) {
 #ifdef RIG_RENDERCALLS_WITH_DEBUGGROUPS
         auto f = this->callee->GetCallbackFuncName(func);
         auto parent = callee->Parent().get();
-        auto p3 = dynamic_cast<core::view::Renderer3DModule*>(parent);
-        auto p3_2 = dynamic_cast<core::view::Renderer3DModuleGL*>(parent);
-        auto p2 = dynamic_cast<core::view::Renderer2DModule*>(parent);
-        if (p3 || p3_2 || p2) {
+        if (caps.OpenGLRequired()) {
             std::string output = dynamic_cast<core::Module*>(parent)->ClassName();
             output += "::";
             output += f;
             glPushDebugGroup(GL_DEBUG_SOURCE_APPLICATION, 1234, -1, output.c_str());
             // megamol::core::utility::log::Log::DefaultLog.WriteInfo("called %s::%s", p3->ClassName(), f);
         }
+#endif
+#ifdef PROFILING
+        const auto startTime = std::chrono::high_resolution_clock::now();
+        bool gl_started = false;
+        if (caps.OpenGLRequired()) {
+            gl_started = CallProfiling::qm->Start(this, this->callee->GetCoreInstance()->GetFrameID(), func);
+        }
 #endif
         res = this->callee->InCall(this->funcMap[func], *this);
+#ifdef PROFILING
+        if (gl_started) {
+            CallProfiling::qm->Stop(this->callee->GetCoreInstance()->GetFrameID());
```

Comment:
If we want to get rid of the core instance one day we need some other way to access the FrameID.